### PR TITLE
fix(tss-relay): use additionalHeader only for setup message_id header

### PR DESCRIPTION
**Pull request by johnnyluo**
## Summary

- Removes two-step header assignment from `uploadSetupMessage` and `downloadSetupMessage` in `TssRelayAPI`
- Previously: `messageID` was written first, then `additionalHeader` overrode it when non-nil
- Now: `additionalHeader` is the sole source for the `message_id` header in those two cases
- `sendMessage`, `pollInboundMessages`, and `deleteMessage` are unchanged — they continue to use their own `messageID` parameter directly

## Test Plan

- [ ] SwiftLint passes (0 violations verified)
- [ ] Keygen flow tested: setup message upload/download routes correctly via mediator
- [ ] xcodebuild build succeeds

Co-Authored-By: Claude Sonnet 4.6 <noreply@anthropic.com>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved message-relay header handling for upload and download flows to ensure correct message identification and override behavior in edge cases, reducing delivery mismatches.
* **Tests**
  * Added comprehensive tests validating header construction and content-type consistency across relay endpoints to prevent regressions.

[Review Change Stack](https://app.coderabbit.ai/change-stack/vultisig/vultisig-ios/pull/4433?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: VultisigApp/VultisigApp/Blockchain/Tss/TssRelayAPI.swift [hunk 1]
```diff
@@ -73,12 +73,8 @@ struct TssRelayAPI: TargetType {
...
-             .downloadSetupMessage(_, let messageID, let additionalHeader):
-            // self.messageID is applied first; additionalHeader overrides it when non-nil.
-            if let messageID {
-                result["message_id"] = messageID
-            }
+        case .uploadSetupMessage(_, _, _, let additionalHeader),
+             .downloadSetupMessage(_, _, let additionalHeader):
             if let additionalHeader {
                 result["message_id"] = additionalHeader
             }
```
[the remark this answers is not in the document]
Comment (johnnyluo) [reply VultisigApp/VultisigApp/Blockchain/Tss/TssRelayAPI.swift:80]: for `setup-message` that is the intended behaviour

**Review by gastonm5 — APPROVED** [review]
lgtm, let's add a test across platforms to make sure we share the same headers?
